lib/connect/tls.js

```diff
@@ -1,12 +1,16 @@
 'use strict'
 var tls = require('tls')
+var net = require('net')
 var debug = require('debug')('mqttjs:tls')
 
 function buildBuilder (mqttClient, opts) {
   var connection
   opts.port = opts.port || 8883
   opts.host = opts.hostname || opts.host || 'localhost'
-  opts.servername = opts.host
+
+  if(net.isIP(opts.host) === 0){
```

Comment:
so can you explain what is happening here in detail? You're checking if the passed in option on host is an IP address? And then the reason for this is because of something to do with TLS SNI?